feat(metrics): allow meter provider to be passed in NewGRPCClient

cojenco opened the pull request:
### Description
- Allow meter provider to be passed in to the Go SDK NewGRPCClient, using the `WithMeterProvider` client option in the Go Storage SDK
- Update monitoring prom tests for gRPC metrics

Depends on https://github.com/googleapis/google-cloud-go/pull/12668


### Link to the issue in case of a bug fix.

### Testing details
1. Manual - Tested
2. Unit tests - NA
3. Integration tests - Added

Manually tested on GKE and GCE. Integration tests are ready once https://github.com/googleapis/google-cloud-go/pull/12668 is launched
```
$ go test ./tools/integration_tests/monitoring/... --integrationTest -v 
=== RUN   TestPromGrpcMetricsSuite
{"timestamp":{"seconds":1755816146,"nanos":897136316},"severity":"INFO","message":"Building GCSFuse from source in the dir: /tmp/gcsfuse_readwrite_test_4022898957 ..."}
{"timestamp":{"seconds":1755816146,"nanos":897185786},"severity":"INFO","message":"Building build_gcsfuse at /tmp/gcsfuse_integration_tests2729044397/build_gcsfuse"}
{"timestamp":{"seconds":1755816151,"nanos":90209635},"severity":"INFO","message":"Building gcsfuse into /tmp/gcsfuse_readwrite_test_4022898957"}
=== RUN   TestPromGrpcMetricsSuite/TestStorageClientGrpcMetrics
--- PASS: TestPromGrpcMetricsSuite (33.62s)
    --- PASS: TestPromGrpcMetricsSuite/TestStorageClientGrpcMetrics (3.24s)
=== RUN   TestPromOTELSuite
{"timestamp":{"seconds":1755816180,"nanos":519715471},"severity":"INFO","message":"Building GCSFuse from source in the dir: /tmp/gcsfuse_readwrite_test_27243481 ..."}
{"timestamp":{"seconds":1755816180,"nanos":519801451},"severity":"INFO","message":"Building build_gcsfuse at /tmp/gcsfuse_integration_tests2133238605/build_gcsfuse"}
{"timestamp":{"seconds":1755816184,"nanos":711409334},"severity":"INFO","message":"Building gcsfuse into /tmp/gcsfuse_readwrite_test_27243481"}
=== RUN   TestPromOTELSuite/TestFsOpsErrorMetrics
=== RUN   TestPromOTELSuite/TestListMetrics
=== RUN   TestPromOTELSuite/TestReadMetrics
=== RUN   TestPromOTELSuite/TestStatMetrics
--- PASS: TestPromOTELSuite (36.87s)
    --- PASS: TestPromOTELSuite/TestFsOpsErrorMetrics (1.60s)
    --- PASS: TestPromOTELSuite/TestListMetrics (1.62s)
    --- PASS: TestPromOTELSuite/TestReadMetrics (1.67s)
    --- PASS: TestPromOTELSuite/TestStatMetrics (1.62s)
PASS
ok      github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/monitoring    70.543s
```

kislaykishore commented on cfg/config_util.go:
```diff
@@ -55,3 +56,14 @@ func ListCacheTTLSecsToDuration(secs int64) time.Duration {
 }
+
+// IsGKEEnvironment returns true if the file descriptor mount option is set.
+func IsGKEEnvironment(mountConfig *Config) bool {
```
The mountPoint string [here](https://github.com/GoogleCloudPlatform/gcsfuse/blob/acdf037408d7969cd398e1481055fae0cdad19c5/cmd/legacy_main.go#L417C40-L417C50) would have prefix /dev/fd/ in GKE environment. See [this](https://github.com/jacobsa/fuse/blob/b8f47b05858b6d8b6ba83df84ce26a7e13442663/unmount_linux.go#L13) for reference.

cojenco replied:
Thanks @kislaykishore  for sharing the code links. Does this mean that the prefix "/dev/fd/" in question cannot be found in the fd mount options `mountConfig.FileSystem.FuseOptions`?

I looked into the linked files. Is the mountPoint string stored in any config struct? What would be the best way to access the `mountPoint` string for this check?

kislaykishore replied:
No, it's not.  Perhaps you can plumb this information using the [createStorageHandle](https://github.com/GoogleCloudPlatform/gcsfuse/blob/980194f078497ef45c7b4a3b2922070ed653c648/cmd/legacy_main.go#L176C24-L176C43) method. Alternatively, see if you could use gcp Resource detector for this.

cojenco replied:
Sounds good, plumbed through the createStorageHandle method and added a flag guard

kislaykishore reviewed (commented):
Overall, looks good. I'll prepare a list of performance tests to run.

cojenco commented:
> Overall, looks good. I'll prepare a list of performance tests to run.

@kislaykishore I have addressed the comments and added a flag. When you have a moment, could you take another pass and share the list of performance tests? 

Also, since the Go storage changes are now in main, would we be able to run the perf and integration tests through CI/CD using those GitHub tags you mentioned?

danielduhh commented:
Hi, is this blocked by something?

kislaykishore commented on internal/storage/storage_handle.go:
```diff
@@ -142,9 +144,17 @@ func createClientOptionForGRPCClient(ctx context.Context, clientConfig *storageu
+	// Pass the OpenTelemetry MeterProvider to the Go storage client,
+	// using the new WithMeterProvider client option.
+	// TODO - detect when running on GKE with clientConfig.IsGKE.
+	if clientConfig.EnableGrpcMetrics {
```
If this is enable-grpc-metrics is false, let's continue the old behavior - `clientOpts = append(clientOpts, storage.WithDisabledClientMetrics())`. If enable-grpc-metrics is true, and isGKE is false, then skip setting `clientOpts = append(clientOpts, storage.WithDisabledClientMetrics())`
